### ISSUE TITLE
Under which conditions may bounding boxes overlap?

### ISSUE THREAD

**Issue by kba** [kba/hocr-spec#23]
In many cases that should not happen (words, lines). For floats, it's inevitable.

hocr-check actually checks for this but it is not spelled out somewhere AFAIK.

**Comment by zuphilip** [comment]
The bounding boxes of lines can overlap and i think this happens easily if the picture is a little skewed and/or letters are going heavily under baseline etc. See for example

![](https://cloud.githubusercontent.com/assets/5199995/16964530/2dd36eea-4dfc-11e6-918e-701020db0fc7.jpg)

**Comment by kba** [comment]
True, in [hocr-check](https://github.com/tmbdev/hocr-tools/blob/master/hocr-check#L57), the assumption is that the overlap should not be more than 20% of the area for lines, paragraphs and careas.

**Comment by zuphilip** [comment]
I think it is interesting to calculate this fraction of overlap, but I don't think it fits into the check, because any number (e.g. 20%) is arbitrary and not really against the specs. Isn't the same routine part of [`hocr-eval-geom`](https://github.com/tmbdev/hocr-tools/blob/master/hocr-eval-geom) and/or [`hocr-eval`](https://github.com/tmbdev/hocr-tools/blob/master/hocr-eval)?

**Comment by zuphilip** [comment]
There `significant_overlap` is 10%.

**Comment by kba** [comment]
Whatever the numbers actually are, it would be helpful to indicate that for typesetting elements, element containment should imply bounding box containment, in most cases. Other elements cannot sensibly have a bounding box (like `ocr_chapter`), that floating elements can overlap containers but should not (or should they?) overlap other contained elements, e.g. an advertisement in a newspaper can cross two `ocr_carea` but should not cross `ocr_line` in those careas, stuff like that.

> I think it is interesting to calculate this fraction of overlap, but I don't think it fits into the check, 

Why not? It must not be a fatal error but it's a helpful measure to evaluate page segmentation.
